jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/NoOpSession.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.apphosting.runtime.jetty9;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpSession;
+import javax.servlet.http.HttpSessionContext;
+
+
+/**
+ * A minimal implementation of HttpSession, just sufficient to keep and id attributes within a
+ * single request.
+ */
+public class NoOpSession implements HttpSession {
+
+  protected ServletContext _context;
+  protected String _clusterId;
+  protected String _nodeId;
+  protected boolean _valid = true;
+  private final Map<String, Object> _attributes = new HashMap<String, Object>();
+
+  /**
+   * @param id
+   * @param nodeId
+   * @param context
+   */
+  protected NoOpSession(String id, String nodeId, ServletContext context) {
+    _clusterId = id;
+    _nodeId = nodeId;
+    _context = context;
+  }
+
+  /**
+   * Finish a request
+   */
+  public void complete() {
```

Comment:
If this class is designed to be sub-classed than `protected` is fine. Otherwise, I would
have it package scope if used by unit-test or private if it doesn't. right now it is `public` and I am not sure why.